doc/source/templates/README.md

```diff
@@ -16,79 +16,129 @@ Coming soon...
+      working_dir: workspace_templates/tests
       python: "3.9"
-      frequency: nightly
+      frequency: nightly-3x
       team: ml
       cluster:
         cluster_env: ../configs/release_test_cluster_env.yaml
-        cluster_compute: ../configs/compute/cpu/aws_small.yaml
+        cluster_compute: ../configs/compute/cpu/aws_release_test.yaml
+
+      variations:
+          - __suffix__: aws
+          - __suffix__: gce
+            env: gce
+            frequency: manual
+            cluster:
+              cluster_env: ../configs/release_test_cluster_env.yaml
+              cluster_compute: ../configs/compute/gpu/gce_release_test.yaml
 
     run:
       timeout: 300
-      script: pip install -U -r requirements.txt
-        && jupyter nbconvert --TagRemovePreprocessor.remove_input_tags='large'
-        --to script --output _test many_model_training.ipynb && ipython _test.py
+      script: jupyter nbconvert --to script --output _test many_model_training.ipynb && ipython _test.py
+    ```
+
+5. (Anyscale Only) To make this template show up as an Anyscale Workspace Template:
+
+    Make a PR on the product repo that links this template as an entry in the
+    [`workspace-templates.yaml` file](https://github.com/anyscale/product/blob/master/backend/workspace-templates.yaml).
+
+    **Use the `anyscale_cluster_env.yaml`, `gce.yaml`, and `aws.yaml` files, NOT the release test counterparts.**
+
+    **Make sure the entry name (e.g., `batch-inference-ray-data`) is unique.**
+
+    ```yaml
+    batch-inference-ray-data:
+      title: Batch Inference
+      description: Scaling Batch Inference with Ray Data
+      path: doc/source/templates/01_batch_inference
+      cluster_env: doc/source/templates/configs/anyscale_cluster_env.yaml
+      compute_config:
+        GCP: doc/source/templates/configs/compute/gpu/gce.yaml
+        AWS: doc/source/templates/configs/compute/gpu/aws.yaml
     ```
 
-    Note: `--TagRemovePreprocessor.remove_input_tags='large'` will make sure that only the small-scale
-    version of the template gets tested nightly.
+6. Success! Your template is ready for review.
+
+<!-- 3. Add an entry to `doc/source/templates/templates.yaml` that links to your template.
+
+    ```yaml
+    - name: Many Model Training using Ray Tune
+      # Paths should be relative to the Ray repo root directory
+      path: doc/source/templates/02_many_model_training
+      cluster_env: doc/source/templates/configs/anyscale_cluster_env.yaml
+      small:
+        compute_config:
+          gcp: doc/source/templates/configs/compute/cpu/gcp_small.yaml
+          aws: doc/source/templates/configs/compute/cpu/aws_small.yaml
+      large:
+        compute_config:
+          # Relative to `path`
+          gcp: doc/source/templates/configs/compute/cpu/gcp_large.yaml
+          aws: doc/source/templates/configs/compute/cpu/aws_large.yaml
+    ```
+
+    When you specify the template's compute config, see `doc/source/templates/configs` for defaults.
```

Comment:
Do we not need this step for the time being?

Reply:
Ohh, good catch. I need to update the guide to match the latest way of doing things (I'm writing this out mostly for myself):

1. Add template to the ray repo `templates.yaml`. CI will check that this is well-formatted, and all linked items exist.
2. Make a PR on the product repo that updates the `backend/workspace-templates.yaml` file to be the exact same as the `templates.yaml` file. This way, all the checks from our CI will apply.
3. Once both PRs are approved, merge them both at the same time.
    - If it's not merged quickly, there may be some time where the links on the product repo are invalid.

Reply:
Got it. Will there be a different solution in the long term? This cross-dependency is probably not sustainable

Reply:
Yeah, I talked to Sofian about moving the source of truth back here to the ray repo.

His concern is Github throttling requests or Github going down affecting workspaces. But, there are some tricks:
1. Caching this file
2. Handling a failed request better in the UI.